tests/python_package_test/test_dask.py

```diff
@@ -1033,6 +1030,84 @@ def collection_to_single_partition(collection):
     client.close(timeout=CLIENT_CLOSE_TIMEOUT)
 
 
+@pytest.mark.parametrize('task', tasks)
+@pytest.mark.parametrize('output', data_output)
+def test_network_params_not_required_but_respected_if_given(client, task, output, listen_port):
+    if task == 'ranking' and output == 'scipy_csr_matrix':
+        pytest.skip('LGBMRanker is not currently tested on sparse matrices')
+
+    if task == 'ranking':
+        _, _, _, _, dX, dy, _, dg = _create_ranking_data(
+            output=output,
+            group=None,
+            chunk_size=10,
+        )
+        dask_model_factory = lgb.DaskLGBMRanker
+    else:
+        _, _, _, dX, dy, _ = _create_data(
+            objective=task,
+            output=output,
+            chunk_size=10,
+        )
+        dg = None
+        if task == 'classification':
+            dask_model_factory = lgb.DaskLGBMClassifier
+        elif task == 'regression':
+            dask_model_factory = lgb.DaskLGBMRegressor
+
+    # rebalance data to be sure that each worker has a piece of the data
+    if output == 'array':
+        dX = dX.persist()
+        dy = dy.persist()
+        _ = wait([dX, dy])
+        if dg is not None:
+            dg = dg.persist()
+            _ = wait(dg)
+        client.rebalance()
+
+    # model 1 - no network parameters givem
+    dask_model1 = dask_model_factory(
+        n_estimators=5,
+        num_leaves=5,
+    )
+    if task == 'ranking':
+        dask_model1.fit(dX, dy, group=dg)
+    else:
+        dask_model1.fit(dX, dy)
+    assert dask_model1.fitted_
+
+    # model 2 - machines given
+    n_workers = len(client.scheduler_info()['workers'])
+    dask_model2 = dask_model_factory(
+        n_estimators=5,
+        num_leaves=5,
+        machines=",".join([
+            "127.0.0.1:" + str(_find_random_open_port())
+            for _ in range(n_workers)
+        ]),
+    )
+    if task == 'ranking':
+        dask_model2.fit(dX, dy, group=dg)
+    else:
+        dask_model2.fit(dX, dy)
+    assert dask_model2.fitted_
```

Comment:
We should really assert that 
1. `assert dask_model2.get_params()['machines'] == machines`
2. check somehow that passed workers from `machines` were used and not any other as it is stated in the test name.

Reply:
oh good point! ok yes, I can do that.

> check somehow that passed workers from machines were used and not any other as it is stated in the test name.

I can test this by binding one of the ports mentioned in `machines`, and asserting that training fails with the "failed to bind port" error

Reply:
ok I think I've addressed both oof these in my most recent commit. I'm glad you mentioned `#1`, because it's something I didn't think about correctly. To fix that, I captured whether `machines` was provided explicitly in parameters, and if it was I had `_train()` preserve it in the fitted model object. Like this

```python
if not machines_in_params:
    for param in _ConfigAliases.get('machines'):
        model._other_params.pop(param, None)
```